### ISSUE TITLE
No card type found

### ISSUE THREAD

**Issue by leopold005** [amaximus/garbage-collection-card#30]
I'm trying to set up this card and I seem to be hitting a brick wall...  I installed the card via HACS and it seems to be all present and correct, but when I start putting in the YAML code for the card, I get a message saying "No card type found" and a further one in orange saying, "Cannot read property 'setConfig' of undefined"

This is the code (after saving) that I have added to Lovelace to configure the card:

```
cards:
  - entity: sensor.blue_bin
    hide_date: false
    icon_color: blue
    icon_size: 35px
    type: 'custom:garbage-collection-card'
resources:
  - type: js
    url: /community_plugin/garbage-collection-card/garbage-collection-card.js
```

I have a sensor called Blue Bin, which I configured using the Garbage Collection integration.

**Comment by amaximus** [comment]
The rendering of your quote/preformatted part is a bit misleading,
You should define in your configuration.yaml:

```
resources:
    - {type: js, url: '/local/community/garbage-collection-card/garbage-collection-card.js'}
```

while in the ui-lovelace.yaml:
```
card:
  - type: 'custom:garbage-collection-card'
    entity: sensor.blue_bin
    hide_date: false
    icon_color: blue
    icon_size: 35px
```

If your configuration.yaml was not needed to be edited, you may just reload your Lovelace UI in the browser (sometimes a cache flush is also needed).

**Comment by leopold005** [comment]
Yeah, I hadn't reckoned on the post being reformatted with bullets.  How annoying!

Anyway, thanks for the help, I got it working now!

**Comment by Liese2007** [comment]
Hi, I have exactly the same problem....seems so.
The plug-in is installed.
My installation folder is:
**\config\www\community\garbage-collection-card**
I added 
```
resources:
  - type: js
    url: /local/community/garbage-collection-card/garbage-collection-card.js
```
and as well
```
resources:
    - {type: js, url: '/local/community/garbage-collection-card/garbage-collection-card.js'}
```
but both return following on the frontend:
```
No card type configured.
card:
  - type: 'custom:garbage-collection-card'
    entity: sensor.stadtreinigung_hamburg_xxxx_blaue_papiertonne
    hide_date: false
    icon_color: blue
    icon_size: 35px
```
The file garbage-collection-card.js is present in that folder.

Any idea, what to check next?

**Comment by amaximus** [comment]
@Liese2007 , since your HASS is presumly running on Windows, you might need to use Windows-style path for the resource.
Please also make sure you clear the browser cache and reload from server (shift-reload).

**Comment by Liese2007** [comment]
Sorry, I copied the path via Samba access.
My HASS is running on a Rasperry Pi with a Hassio image.
Both upper examples should work, if the path itself is ok?

**Comment by amaximus** [comment]
Running on RPi the paths above should be fine.
Please make sure you clear the browser cache and reload from server. Was a fresh new install of the custom card or it worked before?

**Comment by Liese2007** [comment]
My Lovelace looks like this:
```
resources:
  - type: js
    url: /local/community/garbage-collection-card/garbage-collection-card.js
title: UnserHaus
views:
  - badges: []
    cards:
      - entity: binary_sensor.workday_sensor
...... etcpp....
```
- I restarted Home Assistant
- I reloaded (Reload Page)  in Chrome while pressing SHIFT

Still error with this code:
(the sensor state is a valid date - checked in Developer Tools)
```
card:
  - type: 'custom:garbage-collection-card'
    entity: sensor.stadtreinigung_hamburg_xxx_gelbe_wertstofftonne_sack
    hide_date: false
    icon_color: blue
    icon_size: 35px
```
When I want to leave the Manual Card in Lovelace, the error 
**Cannot read property 'setConfig' of undefined**
is shown. But the setup can be saved.

Any more ideas what is wrong?
Greetz
Jörg

**Comment by amaximus** [comment]
Was the card installed via HACS? Looking in a terminal is the $configdir/www/community/garbage-collection-card/garbage-collection-card.js accessible?
Is the garbage-collection-card set into a conditional card? (thus that's why the card: instead of cards: ) If not, then the syntax should be something similar to:

```
cards:
      - type: custom:garbage-collection-card
        entity: sensor.test_garbage_every4w
        icon_color: green
        icon_size: 35px
        hide_date: true
        details_size: 8px
```

**Comment by Liese2007** [comment]
Sorry I am not a command line master...
I am in Terminal & SSH add-on, what do I have to enter to check if the path is available?
`$configdir /www/community/garbage-collection-card/garbage-collection-card.js`
doesn't work

The Card is created lovelace UI 'Manual Card'.

**Comment by amaximus** [comment]
The $configdir would have been referred to the configuration directory of your HomeAssistant. It's just a way of refering to it, not a real variable.
Anyway, since you are using the UI, I guess the path in your resource should be /hacsfiles/garbage-collection-card/garbage-collection-card.js.
This has changed in 0.107.0 (for UI based). yaml based was not affected. Please see:
https://community.home-assistant.io/t/0-107-warning-about-community-plugin/179511

**Comment by Liese2007** [comment]
OK, finally I did it.
I changed in my profile to 'Advanced Option' for the 
Settings --> Lovelace Dashboard.
After that you will see at Lovelace Dashborad the Option to add Resources.
I didn't add any resource line via Lovelace UI (Raw Config).

In addition the basic code to be filled in 'Manual Card' via Lovelace UI looks like:
```
type: 'custom:garbage-collection-card'
entity: sensor.stadtreinigung_hamburg_xxxx_gelbe_wertstofftonne_sack
icon_size: 35px
icon_color: green
hide_date: true
```
The style might have changed in the last HA updates for handling the Lovelace UI.
Thanks for the Card :-)

PS: 
The resource link
`/hacsfiles/garbage-collection-card/garbage-collection-card.js`
I installed via HACS
